core/components/molecules/fileList/FileList.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { BaseProps, extractBaseProps } from '@/utils/types';
+import { FileItem, FileListItem, FileListItemProps } from './FileListItem';
+import { Card } from '@/index';
+export interface FileListProps extends BaseProps {
+  /**
+   * <pre style="font-family: monospace; font-size: 13px; background: #f8f8f8">
+   * Array containing the list of file objects
+   * FileItem:
+   * {
+   *    file: File;
+   *    id?: any;
+   *    status: FileStatus;
+   *    progress: number;
+   *    icon: IconOptions
+   *    errorMessage: string;
+   * }
+   *
+   * | Name | Description | Default |
+   * | --- | --- | --- |
+   * | file | Attached file | |
+   * | id | Unique id/index of file | |
+   * | status | Status of file |
+   * | progress | Progress percent of `Progress Ring` | 0 |
+   * | icon | Name of the icon to be rendered |
+   * | errorMessage | Error Message to be shown when status is `error` | 'Network Error' |
+   * </pre>
+   */
+  fileList: FileItem[];
+  /**
+   * Callback called when file item is clicked
+   */
+  onClick?: (file: File, id?: any) => void;
+
+  actionRenderer?: React.FC<FileListItemProps>;
+}
+
+export const FileList = (props: FileListProps) => {
+  const {
+    fileList,
+    onClick,
+    actionRenderer,
+    className
+  } = props;
+
+  const baseProps = extractBaseProps(props);
+
+  if (fileList.length === 0) return null;
+
+  return (
+    <Card {...baseProps} shadow={'none'} className={className}>
+     {fileList.map((fileItem, index) => (
+        <FileListItem
+          key={index}
+          onClick={onClick}
+          actions={actionRenderer ? actionRenderer(fileItem) : undefined}
```

Comment:
`actionRenderer ? actionRenderer(fileItem) : undefined` can be written as `actionRenderer
 && actionRenderer(fileItem)`